docs/docs/features/dongle.mdx

```diff
@@ -0,0 +1,208 @@
+The recommended approach to doing so (found below) allows you to easily enable a dongle, but it is disabled by default. 
+This is done to ensure that backwards compatibility is maintained for users who do not wish to use a dongle.
+
+### Shield Configuration Files
+
+
+Start by adding the dongle body to the shield configuration.
+
+```kconfig title="Kconfig.shield"
+config SHIELD_MY_KEYBOARD_DONGLE
+    def_bool $(shields_list_contains,my_keyboard_dongle)
+```
+
+### Shield Defconfig Files
+
+Next, add the dongle configuration to your keyboard's `Kconfig.defconfig`.
+
+<Tabs
+defaultValue="unibody"
+values={[
+{label: 'Unibody Keyboard', value: 'unibody'},
+{label: 'Split Keyboard', value: 'split'},
+]}
+>
+<TabItem value="unibody">
+
+```kconfig title="Kconfig.defconfig"
+if SHIELD_MY_KEYBOARD_DONGLE
+
+    config ZMK_KEYBOARD_NAME
+        default "My Board"
+
+    config ZMK_SPLIT_ROLE_CENTRAL
+        default y
+
+    config ZMK_SPLIT
+        default y
+
+    # Prevent the dongle from going to sleep
+    config ZMK_SLEEP
+        default n
+
+    # Increase the transmit power of the dongle
+    config BT_CLTR_TX_PWR_PLUS_8
+        default y
+
+endif
+
+if SHIELD_MY_KEYBOARD
+    config ZMK_KEYBOARD_NAME
+        default "My Board"
+    
+endif
+```
+
+</TabItem>
+<TabItem value="split">
+
+Add the following to your `Kconfig.defconfig` file.
+
+```kconfig title="Kconfig.defconfig"
+if SHIELD_MY_KEYBOARD_DONGLE || SHIELD_MY_KEYBOARD_LEFT || SHIELD_MY_KEYBOARD_RIGHT
+
+    config ZMK_SPLIT
+        default y
+
+endif
+
+if SHIELD_MY_KEYBOARD_DONGLE
+
+    config ZMK_KEYBOARD_NAME
+        default "My Keyboard"
+
+    config ZMK_SPLIT_ROLE_CENTRAL
+        default y
+
+    config ZMK_SPLIT_BLE_CENTRAL_PERIPHERALS
+        default 2
+
+    # Increase the transmit power of the dongle
+    config BT_CLTR_TX_PWR_PLUS_8
+        default y
+
+endif
+```
+
+Modify the `Kconfig.defconfig` file to include the following:
+
+```kconfig title="Kconfig.defconfig"
+
+if SHIELD_MY_KEYBOARD_LEFT
+
+    # Setting this to y will make the left half the central
+    # We want it to be the central when dongle is not used
+    config ZMK_SPLIT_ROLE_CENTRAL
+        default y
+
+    # Check if the left half is the central
+    if ZMK_SPLIT_ROLE_CENTRAL
+        config ZMK_KEYBOARD_NAME
+            default "My Keyboard"
+            
+    endif # ZMK_SPLIT_ROLE_CENTRAL
+
+endif # SHIELD_MY_KEYBOARD_LEFT
+```
+
+</TabItem>
+</Tabs>
+
+### Dongle Overlay File
+
+Replace the `kscan` with `kscan-mock` in the dongle overlay.
+
+:::warning
+Matrix transform must be included here or in the `my_keyboard.dtsi` file.
+:::
+
+```dts title="my_keyboard_dongle.overlay"
+// import the default shield configuration
+#include "my_keyboard.dtsi"
+
+&kscan {
+    compatible = "zmk,kscan-mock";
+
+    columns = <0>;
+    rows = <0>;
+    events = <0>;
+}
+
+```
+
+
+## Enabling the Dongle
+
+Now it is possible to enable the dongle in the user configuration.
```

Comment:
Could you rephrase this so that it reads smoothly for users who are using someone else's keyboard definition and so the upper section doesn't apply to them?